homeassistant/components/geniushub/__init__.py

```diff
@@ -79,16 +100,16 @@ class GeniusBroker:
     def __init__(self, hass, args, kwargs):
         """Initialize the geniushub client."""
         self.hass = hass
-        self.client = hass.data[DOMAIN]["client"] = GeniusHub(
-            *args, **kwargs, session=async_get_clientsession(hass)
-        )
+        self.hub_uid = kwargs.pop(CONF_HUB_UID, None)
+        self.hub_uid = self.hub_uid.upper()
+        self.client = GeniusHub(*args, **kwargs, session=async_get_clientsession(hass))
```

Comment:
I suggest we move the client creation out of the init method and instead pass the client instance to the broker. We don't want any possible side effects in init method.

Reply:
Done.